### DISCUSSION TITLE
Negative impact of --ctrlmidich  on jamulus.ini

### DISCUSSION THREAD

**Discussion by pcar75** [jamulussoftware/jamulus#2678] [Bug Investigation]
In 3.8.2 on Windows 10 ,  I tested that --ctrlmidich  option could be used to show the channel number with the name of users in the client mixerboard ; it does and without needing a midi device.  This could be a facilitator if/when a previous suggestion (Ref.: "Use keyboard shortcuts #2213" ) would be implemented.

However a negative side effect is that the jamulus.ini  file then gets it's fader entries identified with the channel number+name instead of just the name when --ctrlmidich is not used.  Therefore any fader entries with muted, soloed or other adjustments will not be applied to a user in a new session unless the server gives it the same channel number ... which would not happen except by chance.
Since the storing of fader values was intended to be re-applied on future session, I'm pretty sure that this behavior was not intended and is a bug.  

IMHO  the .ini file  fader entries should always be identified with name only and the values applied to the user fader concerned whether or not the --ctrlmidich is used

Examples from a jamulus.ini file :

```
 <storedfadertag5_base64>OTpObyBOYW1l</storedfadertag5_base64>
 <storedfaderlevel5>2</storedfaderlevel5>
 <storedpanvalue5>50</storedpanvalue5>
 <storedfaderissolo5>0</storedfaderissolo5>
 <storedfaderismute5>1</storedfaderismute5>
 <storedgroupid5>-1</storedgroupid5>

```
In Base64 "OTpObyBOYW1l" equals "9:No Name"  i.e. channel number, ":" and name.
It should stay as "Tm8gTmFtZQ==" equals "No Name" i.e name only

I can only presume that the behavior is the same in other environments (MacOS, Linux, etc.).

Thanks for your consideration on this matter.

**Comment by pljones** [comment]
I'd agree - a bug should be raised.  The tag itself should not be changed.  Indeed, the value stored by the client and the value returned by the client should be exactly that set by the "other" client connected to the server.  Where a usage needs the channel number and tag combined, then that should be what's done on a dynamic, case by case, basis.

**Reply by pcar75** [reply]
@pljones   Thanks !
I opened a bug report "Option ctrlmidich writes fader tags in .ini file as Channel':'Name #2680 "

**Comment by ann0see** [comment]
Locked this discussion since everything else should be discussed on #2680
